source/extensions/filters/http/transformation/inja_transformer.cc

```diff
@@ -715,6 +738,23 @@ InjaTransformer::InjaTransformer(const TransformationTemplate &transformation,
   for (auto it = extractors.begin(); it != extractors.end(); it++) {
     extractors_.emplace_back(std::make_pair(it->first, it->second));
   }
+  uint32_t max_size = 4096;
+  if (transformation.max_size() > 0) {
+    max_size = transformation.max_size();
+  }
+
+  const auto &data_sources = transformation.data_sources();
+  for (auto it = data_sources.begin(); it != data_sources.end(); it++) {
+        auto provider_or_error = Envoy::Config::DataSource::DataSourceProvider::create(
```

Comment:
I'm not super familiar with the `Envoy::Config::DataSource::DataSourceProvider::create`. Does this load the data on the data path or pre-load? Doing the former seems potentially expensive.

Reply:
https://github.com/envoyproxy/envoy/blob/release/v1.32/source/common/router/config_impl.cc#L612

Reply:
yes, it watches for file updates